mantidproject/mantid
# Fix and reinstate SANSZOOM system test

### Description of work

#### Summary of work

A bug in `CompareWorkspaces` that caused `NaN` values to be evaluated as equal to finite values meant that system test `SANSZOOMTomlFileConversion.ZOOMV1UserFileTest_m8` was previously passing when it shouldn't have been. When the `CompareWorkspaces` bug was fixed, this test started failing and so was temporarily set to be skipped in PR #38075. This PR fixes the underlying cause of the failure and reinstates the test.



Refs #38088. 




#### Further detail of work

The `SANSZOOMTomlFileConversion` tests are checking that ISIS SANS user files in the new `TOML` format produce the same results as the older `.txt` format. Test `ZOOMV1UserFileTest_m8` uses a `TOML` user file that is supposed to have equivalent settings to the `.txt` user file that's used in `ZOOMLegacyUserFileTest_m8`, and therefore both tests should produce the same reduction output. The settings in the `TOML` format were not completely equivalent, though, so this PR made the following changes to that file:

- Set `detector.correction.position -> rear_z` to `4.5` instead of `-4.5`
- Add a `normalisation.all_monitors` section, as described here: https://docs.mantidproject.org/nightly/techniques/SANS-Toml-v1.html#back-mon-times-t1-t2

### To test:


All CI tests should pass.

I have checked this locally on my Windows machine. The CI tests will check Linux. It's probably not necessary, however if we feel it would be good to check on OSX as well then the reviewer can either do that on a local machine or kick off a build from branch.

*This does not require release notes* because **this is just a change to a system test, so is not user-facing.**



---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
